### ISSUE TITLE
Babel-plugin-styled-components is not compiling my external library

### ISSUE THREAD

**Issue by lucleray** [styled-components/babel-plugin-styled-components#136]
Hello,

I am having issues using `babel-plugin-styled-components` for a specific use case.

I have two folders in a monorepo, that I run using yarn workspaces :
- `app`, with my next.js app
- `ui` , with my 'ui library' using styled-components

I created a git repo that reproduces the bug : https://github.com/lucleray/ssr-ui-library

To reproduce :
- clone the repo
- `yarn install` in the base folder
- `yarn run dev` in `app` folder, `yarn run dev` in `ui` folder
- Open http://localhost:3000 : the app render properly, without error
- Go to `ui/main.js` and change `background: yellow;` to `background: blue;`
- The browser updates properly (the background is blue!)
- I refresh the browser, the background is back to yellow. I also get this error : `Prop className did not match : Server: "sc-bdVaJa hZSvfd" Client: "sc-bdVaJa clQEcn"`

I expect no error. 
I would like to be able to build/watch my 'UI library' and see the updates directly in my app.

I have searched for solutions in the docs, but each of them didn't work :
- declare styled-components as external (it's done in `ui/rollup.config.js`)
- make styled-components a peer dependency

I don't think I'm running two instances of styled-components. 

My guess would be that my UI files are not going through `babel-plugin-styled-components`, but I don't know how to force the plugin to transform them.

Thanks for your help 🙂

**Comment by mxstbr** [comment]
You probably need to add the plugin here? https://github.com/lucleray/ssr-ui-library/blob/master/ui/.babelrc

**Comment by lucleray** [comment]
I just tried to do that, but I still have the same error :
`Prop className did not match. Server: "main__UiButton-s1vypnr9-0 itDLCI" Client: "main__UiButton-s1vypnr9-0 jzgARf"`

The names are matching for the first part, but not the second, i wonder why 🤔

Also, that's the css code generated (after reloading the page) :
```css
/* sc-component-id: main__UiButton-s1vypnr9-0 */
.itDLCI{background:yellow;}.jzgARf{background:green;}
```

When I don't see any error, usually, it looks more like this :
```css
/* sc-component-id: main__UiButton-s1vypnr9-0 */
.main__UiButton-s1vypnr9-0 {} .itDLCI{background:yellow;}
```

Not sure if it's relevant.

**Comment by kopax** [comment]
@lucleray I don't know about your issue but you should not use styled-components as a peer dependency but dependency.

styled-components is not a singleton so there is not reason to have it as a peer.

**Comment by kitten** [comment]
@lucleray I’ll f the first class matches and contains a component name that’s a good sign that the plugin is working as intended.

The second part hints at different CSS in your server and client code. In this case it looks like you might be applying a different background colour before or during React’s hydration.

I’d be happy to help you more but I’d need some code snippets to point this out 😬

**Comment by lucleray** [comment]
@kopax I am making styled-components a peer dependency of my **UI library** only, like recommended here : https://www.styled-components.com/docs/faqs#i-am-a-library-author-should-i-bundle-styledcomponents-with-my-library

@kitten Here is the code : https://github.com/lucleray/ssr-ui-library 🙂

**Comment by kitten** [comment]
@lucleray see my edit :)

@kopax yea, that’s not quite factually correct. We do have singletons in terms of rehydration and accidentally adding SC to your app twice—which is also unnecessary either way—will trigger one instance of SC to “rehydrate” the other ones styles, which will lead to a confusing error

**Comment by lucleray** [comment]
@kitten The repo contains a component that has `background: yellow;`
If you change it (for example to `background: green;`) and refresh the browser, you should see the error.

It seems like in this case, the client gets the new code, but not the server. Could this be related to my webpack server config ?

**Comment by kitten** [comment]
@lucleray yes, that might be the case. From what I can tell everything is working correctly :+1: So this is probably a (hot) reloading issue, I suppose.

**Comment by lucleray** [comment]
@kitten You didn't see any error message in console ?

After you refresh the page, the background color was correct ?

On my side, the color is correct when the hot-reloader updates the page (the hot reloading seem to work fine).

But, once I refresh the page, then I see the error and the color is not correct.

**Comment by kitten** [comment]
@lucleray Sorry, I didn't have time to actually test it, but I'm confident that it's not an issue with this plugin or styled-components itself.

I'd suggest you visit some next.js docs. It might not be reloading your dependencies as expected, assuming that the external dependencies are treated differently 😅

**Comment by dminkovsky** [comment]
Hi @lucleray, did you end up solving this. I am having the same issue as here/https://github.com/styled-components/babel-plugin-styled-components/issues/218: I have yarn workspaces, with a shared workspace for common code between two next.js projects. Styled components in the shared workspace are not properly being handled by babel-plugin-styled-components.

**Comment by dminkovsky** [comment]
Solved this issue with https://stackoverflow.com/a/55748411/741970 (ie. https://github.com/josephluck/next-plugin-custom-babel-config)
